admin/src/main/resources/io/buoyant/admin/js/src/metrics_collector.js

```diff
@@ -27,53 +27,44 @@ define(['jQuery'], function($) {
       _.remove(listeners, function(l) { return l.handler === handler; });
     }
 
-    function generateDeltaPayload(generalData, defaultMetrics, prevMetrics) {
-      var metrics = _(listeners)
-        .flatMap(function(listener){ return listener.metrics(defaultMetrics); })
-        .uniq()
-        .compact()
-        .value();
+    function getTreeDeltaPayload(metricNames, resp, prevResp) {
+      _.each(metricNames, function(metric) {
+        if(_.isArray(metric)) {
+          var prevValue = _.get(prevResp, metric);
+          var currentValue = _.get(resp, metric);
+          var metricCopy = _.clone(metric);
 
-      return _.flatMap(metrics, function(metricName) {
-        var prevValue = prevMetrics[metricName];
-        var currentValue = generalData[metricName];
-        if (prevValue !== undefined && currentValue !== undefined) {
-          return [{
-            name: metricName,
-            delta: currentValue - prevValue,
-            value: currentValue
-          }];
-        } else {
-          return [];
+          if (prevValue !== undefined && currentValue !== undefined) {
+            metricCopy[metricCopy.length - 1] = "delta";
+            _.set(resp, metricCopy, currentValue - prevValue);
+          }
         }
-      })
-
+      });
+      return resp;
     }
 
     return function(initialMetrics) {
       var prevMetrics = initialMetrics;
 
       function update(resp) {
-        var defaultMetrics = _.keys(resp);
-        var specific = generateDeltaPayload(resp, defaultMetrics, prevMetrics);
+        var metricsToGet = _.flatMap(listeners, function(l) { return l.metrics(resp); });
+        var metricsWithDeltas = getTreeDeltaPayload(metricsToGet, resp, prevMetrics);
```

Comment:
Given the new API, I think we should take out the `.metrics()` function that listeners define and instead calculate deltas for every counter in the payload. I don't think it'd be a performance hit and it'd certainly be easier to maintain